dozer/cogs/tba.py

```diff
@@ -37,100 +41,268 @@ def __init__(self, bot):
     """
 
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def eventsfor(self, ctx, team_num: int, year: int = None):
+        """Get the events a team is registered for a given year. Defaults to current (or upcoming) year."""
+        try:
+            events = await self.session.team_events(team_num, year=year)
+            # will fall back to the current year
+            year = year or (await self.session.status()).current_season
+        except aiotba.http.AioTBAError:
+            raise BadArgument(f"Couldn't find matching data!")
+
+        if not events:
+            raise BadArgument(f"Couldn't find matching data!")
+
+        e = discord.Embed(color=discord.Color.blurple())
+
+        e.add_field(name=f"Registered events for FRC Team {team_num} in {year}:",
+                    value="\n".join(i.name for i in events))
+        await ctx.send(embed=e)
+
+    eventsfor.example_usage = """
+    `{prefix}tba eventsfor 1533` - show the currently registered events for team 1533, Triple Strange
+    """
+
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def media(self, ctx, team_num: int, year: int = None):
+        """Get media of a team for a given year. Defaults to current year."""
+        if year is None:
+            year = datetime.datetime.today().year
+        try:
+            team_media = await self.session.team_media(team_num, year)
+
+            pages = []
+            base = f"FRC Team {team_num} {year} Media: "
+            for media in team_media:
+
+                if media.type == "youtube":
+                    pages.append(f"**{base} YouTube** \nhttps://youtu.be/{media.foreign_key}")
+                    continue
+                else:
+                    name, url, img_url = {
+                        "cdphotothread": (
+                            "Chief Delphi",
+                            "https://www.chiefdelphi.com/media/photos/{foreign_key}",
+                            "https://www.chiefdelphi.com/media/img/{image_partial}"
+                        ),
+                        "imgur": (
+                            "Imgur",
+                            "https://imgur.com/{foreign_key}",
+                            "https://i.imgur.com/{foreign_key}.png"
+                        ),
+                        "instagram-image": (
+                            "instagram",
+                            "https://www.instagram.com/p/{foreign_key}",
+                            "https://www.instagram.com/p/{foreign_key}/media"
+                        ),
+                        "grabcad": (
+                            "GrabCAD",
+                            "https://grabcad.com/library/{foreign_key}",
+                            "{model_image}"
+                        )
+                    }.get(media.type, (None, None, None))
+                    if name is None:
+                        # print("Whack media", media.__dict__, "unprocessed")
```

Comment:
This looks like it should've been removed.

Reply:
there's exactly one type of media that will trigger that code, and it's the power up team icons iirc

Reply:
The `external-link` type will as well. 3132's media in 2017 has two PDF links for example.